Utils/auto_update_docker/extract_affected_content_items.py

```diff
@@ -0,0 +1,447 @@
+            )
+        else:
+            logging.info(f"No batch config was found for {docker_image = }, and {current_batch_index = }")
+
+        if not affected_content_items["content_items"]:
+            logging.info(f"{docker_image = } does not have any content items to update")
+        affected_content_items_by_docker_image[docker_image] = affected_content_items
+    return affected_content_items_by_docker_image
+
+
+def calculate_affected_docker_images(
+    docker_images_arg: str,
+    images_to_exclude: list[str],
+    all_docker_images: list[str],
+) -> list[str]:
+    """Calculates the docker images that will be used in the current batch.
+    Docker images in the 'images_to_exclude' list will ALWAYS be excluded.
+    Args:
+        docker_images_arg (str): Docker images arg supplied by the user. This will either be:
+            i) ALL - Use all docker images\n
+            ii) docker1,docker2,... - A list of docker images to use.\n
+            iii) ALL@docker1,docker2,... - All docker images, excluding docker1,docker2,...\n
+        images_to_exclude (list[str]): A list of images that will be excluded.
+        all_docker_images (list[str]): All docker images returned from the graph.
+
+    Returns:
+        list[str]: A list of docker images that will be used in the current batch.
+    """
+    images_without_excluded_ones = set(all_docker_images) - set(images_to_exclude)
+    if docker_images_arg == "ALL":
+        return list(images_without_excluded_ones)
+    # TODO Update to @ in Confluence and Jira ticket
+    images_args = docker_images_arg.split("@")
+    if len(images_args) == 1:
+        # Comma separated list case
+        specific_images = images_args[0].split(",")
+        return list(images_without_excluded_ones.intersection(set(specific_images)))
+    elif len(images_args) == 2:
+        # All/docker1,docker2 case
+        specific_images_to_exclude = images_args[1].split(",")
+        return list(images_without_excluded_ones - set(specific_images_to_exclude))
+    else:
+        logging.error("Wrong docker images args")
+        return []
+
+
+def query_used_dockers_per_content_item(tx: Transaction) -> list[tuple[str, str, bool, str, str]]:
+    """
+    Queries the content graph for the following data:
+    1. Docker image.
+    2. Path of the content items.
+    3. If content item is configured for auto updating its docker image.
+    4. Pack path.
+    5. Pack support
+    """
+    return list(
+        tx.run(
+            """
+            MATCH (pack:Pack) <-[:IN_PACK] - (iss)
+            WHERE iss.content_type IN ["Integration", "Script"]
+            AND NOT iss.deprecated
+            AND NOT iss.type = 'javascript'
+            AND NOT pack.support = 'community'
+            AND NOT pack.object_id = 'ApiModules'
+            AND iss.docker_image IS NOT NULL
+            AND NOT pack.hidden
+            Return iss.docker_image, iss.path, iss.auto_update_docker_image, pack.path, pack.support
```

Comment:
filter out acc to  `auto_update_docker_image` in the query instead of returning it and filtering in the code